src/Features/Core/Portable/Diagnostics/HostAnalyzerManager.cs

```diff
@@ -165,6 +165,14 @@ public bool IsAnalyzerSuppressed(DiagnosticAnalyzer analyzer, Project project)
                 return false;
             }
 
+            // If user has disabled analyzer execution for this project, we only want to execute required analyzers
+            // that report diagnostics with category "Compiler".
+            if (!project.State.RunAnalyzers &&
+                GetDiagnosticDescriptors(analyzer).All(d => d.Category != DiagnosticCategory.Compiler))
```

Comment:
I am in two minds here. I am special casing "Add Usings" and "Remove unncessary usings" analyzers so they execute them even if user has turned off analyzer execution. These are inherently compiler diagnostics, former is essentially a workaround for https://github.com/dotnet/roslyn/issues/32628, and the latter for https://github.com/dotnet/roslyn/issues/28773 (to allow special treatment in generated code files). Additionally, these two analyzers (and their corresponding fixers) are sort of important for good editing experience. We can either take the approach that I have taken here (consider analyzer diagnostics with category "compiler" as equivalent to compiler diagnostics) OR hardcode the analyzer types themselves and handle them specially. I felt the former is a clearer approach, but open to better suggestions.

Reply:
I like the current approach. My only concern is there's currently no way to prevent 3rd party authors to use "Compiler" as category.

---
In reply to: [339351992](https://github.com/dotnet/roslyn/pull/39544#discussion_r339351992) [](ancestors = 339351992)

Reply:
For the initial implementation, we should follow the explicit user request. There should not be hard-coded exceptions. We can run the compiler, but we cannot run the IDE0005 analyzer.

Reply:
I think that makes the experience extremely unpleasant, especially given these two specialized analyzers are just unfortunate workarounds for compiler functionality. If you don't like this approach, I can hard code and special the 2 analyzer names here.